host-controller/src/main/java/org/jboss/as/host/controller/model/jvm/JvmOptionsBuilderFactory.java

```diff
@@ -177,17 +177,17 @@ boolean checkAdditionalJvmOption(String option){
 
     private static class OracleJvmOptionsBuilder extends JvmOptionsBuilder {
         private static final int JVM_MAJOR_VERSION = JvmElement.getJVMMajorVersion();
-        private static final List<String> ALLOWED_JDK9_PARAMS = Arrays.asList("--permit-illegal-access", "--add-exports", "--add-opens", "--add-modules", "--add-reads", "--illegal-access");
+        private static final List<String> ALLOWED_MODULAR_JDK_PARAMS = Arrays.asList("--add-exports", "--add-opens", "--add-modules", "--add-reads", "--illegal-access");
 
         private OracleJvmOptionsBuilder(JvmType type) {
             super(type);
         }
 
         @Override
         boolean checkAdditionalJvmOption(String option) {
-            if (JVM_MAJOR_VERSION >= 9 ) {//when java 11 comes, make sure this params are still fine.
+            if (JVM_MAJOR_VERSION >= 9 ) {
                 return true; //all params are fine
-            }else if (ALLOWED_JDK9_PARAMS.contains(option.contains("=") ? option.substring(0, option.indexOf("=")) : option)){ //drop jdk9 specific params on jdk 8
+            } else if (ALLOWED_MODULAR_JDK_PARAMS.contains(option.contains("=") ? option.substring(0, option.indexOf("=")) : option)){ //drop jdk9 specific params on jdk 8
                 return false;
             }
```

Comment:
Thinking about it more, none of this from L188 is correct. Wasn't correct before either. It is using the JVM version of the HC to decide what's legal for the server, and in this case that is not a safe algorithm.

A JDK 11 HC can launch a JDK 8 server.

This kind of logic is only ok in cases where either we know the server is running the same VM version as the HC.  Otherwise we have to know what the VM the server is running is.